src/get_indexing.F90

```diff
@@ -112,44 +112,42 @@ subroutine Get_Indexing(Ctrl, SAD_Chan, SPixel, MSI_Data, status)
    ! Initialise Approach and Class bits
    SPixel%variables_retrieved = (Ctrl%Approach-1) + 8*(Ctrl%Class-1)
 
-
-   ! If the views have different illumination conditions then skip this pixel
-   do i_view = 1, Ctrl%Ind%NViews
-      SPixel%illum(i_view) = MSI_Data%illum(SPixel%Loc%X0, SPixel%Loc%Y0, i_view)
-
-      if ((i_view > 1) .and. (.not. Ctrl%all_channels_same_view)) then
-         if (SPixel%illum(i_view - 1) /= SPixel%illum(i_view)) then
-            status = SPixelIndexing
-            return
-         end if
-      end if
-   end do
-
    ! Force nighttime retrieval
    if (Ctrl%force_nighttime_retrieval) then
-      SPixel%Illum(:) = INight
+      SPixel%Illum = INight
+   else
+      ! If the views have different illumination conditions then skip this pixel
+      do i_view = 1, Ctrl%Ind%NViews
+         SPixel%Illum = MSI_Data%illum(SPixel%Loc%X0, SPixel%Loc%Y0, i_view)
+         if ((i_view > 1) .and. (.not. Ctrl%all_channels_same_view)) then
+            if (MSI_Data%Illum(SPixel%Loc%X0, SPixel%Loc%Y0, i_view-1) /= SPixel%Illum) then
+               status = SPixelIndexing
+               return
+            end if
+         end if
+      end do
    end if
 
    ! Find which channels are either not used based on illumination conditions
    ! and/or are flagged as missing.
    is_not_used_or_missing = .false.
 
-   if (SPixel%Illum(1) .eq. IDay) then
+   if (Spixel%Illum .eq. IDay) then
```

Comment:
Did you intend to lower-case P for the rest of this file?

Reply:
Ah good catch. No I did not intend to make the 'p' in `SPixel%Illum` lower case. That's a typo. I've reverted all instances of `Spixel%Illum` to `SPixel%Illum`